### PR TITLE
Check permission before updating language

### PR DESCRIPTION
**Pull request by zkhin**
Fixes: https://github.com/SeedCompany/cord-api-v3/issues/979

### CODE REVIEW

File: src/components/language/language.service.ts [hunk 1]
```diff
@@ -200,6 +202,29 @@ export class LanguageService {
...
+      .query()
+      .call(matchRequestingUser, session)
+      .call(matchUserPermissions, label, id)
+      .return('permList, node')
+      .first();
+
+    if (!result) {
+      throw new ForbiddenException(
+        'User does not have permission to read a language'
+      );
```
Comment (CarsonF) [inline src/components/language/language.service.ts:224]: Read or update?

Comment (zkhin) [reply src/components/language/language.service.ts:224]: Yeah, update!

Comment (CarsonF) [reply src/components/language/language.service.ts:224]: You aren't checking for either of these permissions though. How do you know which one?

Comment (CarsonF) [reply src/components/language/language.service.ts:224]: Same ^

Comment (zkhin) [reply src/components/language/language.service.ts:224]: So I have to check the permissions(`canEdit`) fields one by one? I just checked the `permList` and if it's empty array, it throws an exception.

Comment (michaelmarshall) [reply src/components/language/language.service.ts:224]: @zkhin try these changes on lines 490+ 498. That should do it. Let me know.
image

Comment (zkhin) [reply src/components/language/language.service.ts:224]: After only these changes, we will get `Could not find object` 404 exceptions if the user does not have permission. Is this fine? And I don't need to check permissions before updating `Language`?

Comment (michaelmarshall) [reply src/components/language/language.service.ts:224]: yes that is fine for now. after mvp we will refactor to give the FE better errors.

Comment (zkhin) [reply src/components/language/language.service.ts:224]: Okay, cool

**Review by CarsonF — DISMISSED** [review]
If it's good with @michaelmarshall it's good with me! 👍

**Pull request merged**